elastic/kibana
# [Security Solution][Endpoint] Cypress test to validate that Endpoint can stream alerts to ES/Kibana

## Summary

- Adds cypress test that stands up a real endpoint and validates that it can trigger alerts and send those to ES/Kbn and that they show up on the Alerts list

### Testing

The test was added to a Test Suite that is not currently running under CI. To run it manually:

- First, ensure that you have Docker and [Multipass](https://multipass.run/) installed
- Ensure that you have Kibana Running locally with Fleet setup with Fleet-server (Tip: you can execute the `run_endpoint_agent.js` script, which will setup fleet server locally with Docker)
- Run tests (see command below. Adjust env. variable values if values are different in your local env.)

```
cd x-pack/plugins/security_solution

CYPRESS_BASE_URL=http://localhost:5601 \
CYPRESS_KIBANA_URL=http://localhost:5601 \
CYPRESS_ELASTICSEARCH_USERNAME=elastic \
CYPRESS_ELASTICSEARCH_PASSWORD=changeme \
CYPRESS_ELASTICSEARCH_URL=http://localhost:9200 yarn cypress:dw:endpoint:open
```
